### DISCUSSION TITLE
Email notification do not work

### DISCUSSION THREAD

**Discussion by EsmailELBoBDev2** [upptime/upptime#526] [Q&A] [answered]
So I followed the documentation about setup a mailgun SMTP and it just does not work. 

In mailgun, there are no logs (so no emails being sent)
![image](https://user-images.githubusercontent.com/28893833/156919871-ac36d463-439b-4d5f-9c99-3eceedefb81d.png)

And in my repo's secrets I already added the values:
![image](https://user-images.githubusercontent.com/28893833/156919912-9551f316-1611-4cd8-85b2-c1f6279a4905.png)

and you can check my config in https://github.com/EsmailELBoBDev2/upptime/blob/master/.upptimerc.yml 
___
So yes when website goes down I do not get any notifications

PS: I know mailgun `from email` must be same as your `smtp username` value so I have already done that! and yes I already added my `to email` in mailgun's sandbox

**Comment by EsmailELBoBDev2** [comment]
@AnandChowdhary @Julien-Osman any help?

**Reply by AnandChowdhary** [reply]
It doesn't look like the environment variable names are correct: https://upptime.js.org/docs/notifications#email

**Reply by EsmailELBoBDev2** [reply]
@AnandChowdhary It is same copy paste ??
![image](https://user-images.githubusercontent.com/28893833/163975274-32b4cdb8-4442-4d6f-b94c-c3cad55d3bf7.png)

**Reply by AnandChowdhary** [reply]
That's right, but on the previous screenshot you have environment variants such as "SMTPPASSWORD". I see now that it looks right! Can you confirm that the indentation in YAML is correct? Since that makes a difference in YAML (i.e., you're using 2 spaces above but four below). You can delete the "notifications" key in YAML since that's probably confusing it; all notification data should directly be set as environment variables.

**Reply by EsmailELBoBDev2** [reply]
@AnandChowdhary oh yes my first screenshot was wrong and oh you mean like this? 
https://github.com/EsmailELBoBDev2/upptime/commit/b3939eaf54ae28838fa27a29c2fa55628b513c18

**Reply by AnandChowdhary** [reply]
You can delete all of those from the .upptimerc.yml and just add those environment variables in the repository settings on GitHub.

**Reply by EsmailELBoBDev2** [reply]
Okay, excuse me for my dumbness I swear to god I know how computer works 😂

You mean like this: https://github.com/EsmailELBoBDev2/upptime/commit/6a1201f80b6024d16a293977fa854a46c64e625f ?

PS: my secrets are: 
![image](https://user-images.githubusercontent.com/28893833/164192415-68593408-e61f-4004-8853-c4a4019c7105.png)

**Reply by EsmailELBoBDev2** [reply]
UPDATE: oh it works!!! but in temp files it says it gets updated daily so I think I need to add my env vars inside upptimerc so how to fix it?

**Reply by AnandChowdhary** [reply]
Congrats, awesome that it works! No, you don't need to add the environment variables to .upptimerc.yml, since that file is reserved for configuration and we directly use environment variables for notifications.

**Reply by EsmailELBoBDev2** [reply]
but my file will get overwritten (someone do not read comments) so how to solve this then?? I cant add it in my config file upptimerc and I can't add it inside workflow as it will get overwritten by the bot so how to solve it

**Reply by AnandChowdhary** [reply]
I don't understand -- why would your file get overwritten? The automated update will only change the .github/workflows files and I'm assuming you didn't make any changes there, right?

**Reply by EsmailELBoBDev2** [reply]
Umm that's exactly what I changed to make it work 😂 I went to `uptime.yml` in workflows and added them in env var.

I'm trying crazy thing where I add env vars inside `.upptimerc.yml`

**Reply by AnandChowdhary** [reply]
Aah ok, so here's what you do -- just add the environment variables in the GitHub UI, nowhere else. Not in the config, not in the workflows, nowhere.

**Reply by EsmailELBoBDev2** [reply]
Oh...Like this? 
![image](https://user-images.githubusercontent.com/28893833/164200657-b87b3492-3468-462e-b068-fb2c08e20188.png)


After we done expect a PR for notification wikipage explaining step by step how to do it. It's really hard to follow up 😂

**Reply by AnandChowdhary** [reply]
No, not "Environments", go to "Secrets" -> "Actions" (/settings/secrets/actions), and then add them like you did before. To enable email notifications, you need these: https://upptime.js.org/docs/notifications#email. Can you confirm that you have all three?

**Reply by EsmailELBoBDev2** [reply]
I feel I'm so stupid like now 😂

So wait if I get it right, I add them like I did with my secrets? Like env name and in value I add True for example?
![image](https://user-images.githubusercontent.com/28893833/164202185-ec50fdd1-63d3-430b-bf5a-dceea8b18160.png)

**Reply by AnandChowdhary** [reply]
Yes, that's exactly right. It should work now?

**Reply by AnandChowdhary** [reply]
If you want to use email-based notifications via SMTP, you also need these: https://upptime.js.org/docs/notifications/#smtp

**Reply by EsmailELBoBDev2** [reply]
yup already set those too and now testing...

**Reply by EsmailELBoBDev2** [reply]
It works :tada: so here you go: https://github.com/upptime/upptime.js.org/pull/27 not much edit but certainly more easy to understand

**Comment by AnandChowdhary** [comment]
You only need environment variables for notifications, removing them from .upptimerc will fix everything! :)

**Comment by EsmailELBoBDev2** [comment, marked as answer]
You need to add these values as your github secrets not as config part of your config file. 

So it goes like:
Environment variable -> Secret name
Environment Value -> Secret value

___

*Thanks to @AnandChowdhary*
